File: javascript/gigasecond/gigasecond.js
const Gigasecond = function Gigasecond(start) {
  this.start = start;
};

Gigasecond.prototype.date = function date() {
  /* 1e9 seconds * 1000 milliseconds/second */
  return new Date(this.start.getTime() + 1e9*1000);
};

module.exports = Gigasecond;

File: ruby/gigasecond/gigasecond.rb
# frozen_string_literal: true

# Calculate date a gigasecond away from input
class Gigasecond
  GIGASECOND = 10**9
  def self.from(time)
    # According to documentation, you can add a number and it will add that many
    # seconds.
    # https://ruby-doc.org/core-2.3.3/Time.html
    time + GIGASECOND
  end
end
